<a href="https://colab.research.google.com/github/DaveChui/Prep_Group-4-Project/blob/main/DS14_GROUP_IV_PROJECT_2022_02_CRYPTO_ANALYSIS.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#1.0 IMPORTING THE LIBRARIES

In [5]:
# Importing the pandas library
import pandas as pd

# Importing the numpy library
import numpy as np

from functools import reduce

#2.0 DATA LOADING AND CLEANING.

In [6]:
#First load the bitcoin data and preview records 
bitcoin_data = pd.read_csv('bitcoin.csv')
bitcoin_data.head()

Date         Open         High  ...        Close    Adj Close      Volume
0  2017-11-21  8205.740234  8348.660156  ...  8071.259766  8071.259766  4277609984
1  2017-11-22  8077.950195  8302.259766  ...  8253.549805  8253.549805  3633530112
2  2017-11-23  8232.379883  8267.400391  ...  8038.770020  8038.770020  4225179904
3  2017-11-24  8074.020020  8374.160156  ...  8253.690430  8253.690430  5058610176
4  2017-11-25  8241.709961  8790.919922  ...  8790.919922  8790.919922  4342060032

[5 rows x 7 columns]

In [7]:
#Load the ethereum data and preview records 
ethereum_data = pd.read_csv('ethereum.csv')
ethereum_data.head()

Date        Open        High  ...       Close   Adj Close      Volume
0  2017-11-21  367.442993  372.470001  ...  360.401001  360.401001   949912000
1  2017-11-22  360.312012  381.420013  ...  380.652008  380.652008   800819008
2  2017-11-23  381.438995  425.548004  ...  410.165985  410.165985  1845680000
3  2017-11-24  412.501007  480.972992  ...  474.911011  474.911011  2292829952
4  2017-11-25  475.675995  485.191986  ...  466.276001  466.276001  1422080000

[5 rows x 7 columns]

In [8]:
#Load the tether data and preview records 
tether_data = pd.read_csv('tether.csv')
tether_data.head()

Date     Open     High       Low    Close  Adj Close      Volume
0  2017-11-09  1.01087  1.01327  0.996515  1.00818    1.00818   358188000
1  2017-11-10  1.00650  1.02423  0.995486  1.00601    1.00601   756446016
2  2017-11-11  1.00598  1.02621  0.995799  1.00899    1.00899   746227968
3  2017-11-12  1.00602  1.10591  0.967601  1.01247    1.01247  1466060032
4  2017-11-13  1.00448  1.02929  0.975103  1.00935    1.00935   767884032

In [9]:
#Getting information about the datasets ie datatypes, number of entries,columns
bitcoin_data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1554 entries, 0 to 1553
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Date       1554 non-null   object 
 1   Open       1554 non-null   float64
 2   High       1554 non-null   float64
 3   Low        1554 non-null   float64
 4   Close      1554 non-null   float64
 5   Adj Close  1554 non-null   float64
 6   Volume     1554 non-null   int64  
dtypes: float64(5), int64(1), object(1)
memory usage: 85.1+ KB


In [10]:
ethereum_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1554 entries, 0 to 1553
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Date       1554 non-null   object 
 1   Open       1554 non-null   float64
 2   High       1554 non-null   float64
 3   Low        1554 non-null   float64
 4   Close      1554 non-null   float64
 5   Adj Close  1554 non-null   float64
 6   Volume     1554 non-null   int64  
dtypes: float64(5), int64(1), object(1)
memory usage: 85.1+ KB


In [11]:
tether_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1566 entries, 0 to 1565
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Date       1566 non-null   object 
 1   Open       1566 non-null   float64
 2   High       1566 non-null   float64
 3   Low        1566 non-null   float64
 4   Close      1566 non-null   float64
 5   Adj Close  1566 non-null   float64
 6   Volume     1566 non-null   int64  
dtypes: float64(5), int64(1), object(1)
memory usage: 85.8+ KB


From the above info we'll have to change the Date column from object data type to datetime.

In [12]:
#Changing the bitcoin Date column's data type from object to date/time.
bitcoin_data['Date'] = pd.to_datetime(bitcoin_data['Date'])
#displaying results
bitcoin_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1554 entries, 0 to 1553
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype         
---  ------     --------------  -----         
 0   Date       1554 non-null   datetime64[ns]
 1   Open       1554 non-null   float64       
 2   High       1554 non-null   float64       
 3   Low        1554 non-null   float64       
 4   Close      1554 non-null   float64       
 5   Adj Close  1554 non-null   float64       
 6   Volume     1554 non-null   int64         
dtypes: datetime64[ns](1), float64(5), int64(1)
memory usage: 85.1 KB


In [13]:
#Changing the ethereum Date column's data type from object to date/time.
ethereum_data['Date'] = pd.to_datetime(ethereum_data['Date'])
#displaying results
ethereum_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1554 entries, 0 to 1553
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype         
---  ------     --------------  -----         
 0   Date       1554 non-null   datetime64[ns]
 1   Open       1554 non-null   float64       
 2   High       1554 non-null   float64       
 3   Low        1554 non-null   float64       
 4   Close      1554 non-null   float64       
 5   Adj Close  1554 non-null   float64       
 6   Volume     1554 non-null   int64         
dtypes: datetime64[ns](1), float64(5), int64(1)
memory usage: 85.1 KB


In [14]:
#Changing the tether Date column's data type from object to date/time.
tether_data['Date'] = pd.to_datetime(tether_data['Date'])
#displaying the results
tether_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1566 entries, 0 to 1565
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype         
---  ------     --------------  -----         
 0   Date       1566 non-null   datetime64[ns]
 1   Open       1566 non-null   float64       
 2   High       1566 non-null   float64       
 3   Low        1566 non-null   float64       
 4   Close      1566 non-null   float64       
 5   Adj Close  1566 non-null   float64       
 6   Volume     1566 non-null   int64         
dtypes: datetime64[ns](1), float64(5), int64(1)
memory usage: 85.8 KB


In [15]:
#Ensure uniformity by changing the column names to lower case, stripping of the whitespaces,removing periods, 
#this ensures good readerbility plus every possible naming issue is covered.
#This is to be applied to the bitcoin, tether and ethereum datasets.
#Bitcoin dataset

bitcoin_data.columns = bitcoin_data.columns.str.lower().str.strip().str.replace(' ', '_').str.replace('.', '')
bitcoin_data.columns

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:6: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  


Index(['date', 'open', 'high', 'low', 'close', 'adj_close', 'volume'], dtype='object')

In [16]:
#Ethereum dataset
ethereum_data.columns = ethereum_data.columns.str.lower().str.strip().str.replace(' ', '_').str.replace('.', '')
ethereum_data.columns

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  


Index(['date', 'open', 'high', 'low', 'close', 'adj_close', 'volume'], dtype='object')

In [17]:
#Tether dataset
tether_data.columns = tether_data.columns.str.lower().str.strip().str.replace(' ', '_').str.replace('.', '')
tether_data.columns

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:2: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  


Index(['date', 'open', 'high', 'low', 'close', 'adj_close', 'volume'], dtype='object')

In [18]:
#Renaming the columns in both our datasets to differentiate between the bitcoin and ethereum columns by adding an 'eth' suffix for ethereum
#and a 'bit' for bitcoin.
bitcoin_data2 = bitcoin_data.add_suffix('bit')
#previewing results
bitcoin_data2.head()

datebit      openbit      highbit  ...     closebit  adj_closebit   volumebit
0 2017-11-21  8205.740234  8348.660156  ...  8071.259766   8071.259766  4277609984
1 2017-11-22  8077.950195  8302.259766  ...  8253.549805   8253.549805  3633530112
2 2017-11-23  8232.379883  8267.400391  ...  8038.770020   8038.770020  4225179904
3 2017-11-24  8074.020020  8374.160156  ...  8253.690430   8253.690430  5058610176
4 2017-11-25  8241.709961  8790.919922  ...  8790.919922   8790.919922  4342060032

[5 rows x 7 columns]

In [19]:
#repeat the process for ethereum dataset adding "eth"

ethereum_data2 = ethereum_data.add_suffix('eth')
#previewing results
ethereum_data2.head()

dateeth     openeth     higheth  ...    closeeth  adj_closeeth   volumeeth
0 2017-11-21  367.442993  372.470001  ...  360.401001    360.401001   949912000
1 2017-11-22  360.312012  381.420013  ...  380.652008    380.652008   800819008
2 2017-11-23  381.438995  425.548004  ...  410.165985    410.165985  1845680000
3 2017-11-24  412.501007  480.972992  ...  474.911011    474.911011  2292829952
4 2017-11-25  475.675995  485.191986  ...  466.276001    466.276001  1422080000

[5 rows x 7 columns]

In [20]:
#repeat the process for tether dataset adding "eth"

tether_data2 = tether_data.add_suffix('tet')
#previewing results
tether_data2.head()

datetet  opentet  hightet    lowtet  closetet  adj_closetet   volumetet
0 2017-11-09  1.01087  1.01327  0.996515   1.00818       1.00818   358188000
1 2017-11-10  1.00650  1.02423  0.995486   1.00601       1.00601   756446016
2 2017-11-11  1.00598  1.02621  0.995799   1.00899       1.00899   746227968
3 2017-11-12  1.00602  1.10591  0.967601   1.01247       1.01247  1466060032
4 2017-11-13  1.00448  1.02929  0.975103   1.00935       1.00935   767884032

In [21]:
#renaming the 'dateeth' column to 'datebit' to be able to merge the two datasets since the data was collected in the same period.

ethereum_data3 = ethereum_data2.rename(columns={'dateeth': 'datebit'}, inplace=True)


In [22]:
#renaming the 'dateeth' column to 'datebit' to be able to merge the two datasets since the data was collected in the same period
tether_data3 = tether_data2.rename(columns={'datetet': 'datebit'}, inplace=True)


In [23]:
tether_data2.head()

datebit  opentet  hightet    lowtet  closetet  adj_closetet   volumetet
0 2017-11-09  1.01087  1.01327  0.996515   1.00818       1.00818   358188000
1 2017-11-10  1.00650  1.02423  0.995486   1.00601       1.00601   756446016
2 2017-11-11  1.00598  1.02621  0.995799   1.00899       1.00899   746227968
3 2017-11-12  1.00602  1.10591  0.967601   1.01247       1.01247  1466060032
4 2017-11-13  1.00448  1.02929  0.975103   1.00935       1.00935   767884032

In [24]:
tether_data2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1566 entries, 0 to 1565
Data columns (total 7 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   datebit       1566 non-null   datetime64[ns]
 1   opentet       1566 non-null   float64       
 2   hightet       1566 non-null   float64       
 3   lowtet        1566 non-null   float64       
 4   closetet      1566 non-null   float64       
 5   adj_closetet  1566 non-null   float64       
 6   volumetet     1566 non-null   int64         
dtypes: datetime64[ns](1), float64(5), int64(1)
memory usage: 85.8 KB


#Merging the datasets

In [25]:
# creating a new data set "crypto_data" by combining the Bitcoin, Tether and Ethereum datasets and displaying it.
#crypto_data = pd.merge(bitcoin_data2, ethereum_data2, left_on='datebit', right_on='datebit')
#crypto_data

crypto_df = bitcoin_data2.merge(tether_data2,on='datebit').merge(ethereum_data2,on='datebit')

In [26]:
#Checking the info again
crypto_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1554 entries, 0 to 1553
Data columns (total 19 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   datebit       1554 non-null   datetime64[ns]
 1   openbit       1554 non-null   float64       
 2   highbit       1554 non-null   float64       
 3   lowbit        1554 non-null   float64       
 4   closebit      1554 non-null   float64       
 5   adj_closebit  1554 non-null   float64       
 6   volumebit     1554 non-null   int64         
 7   opentet       1554 non-null   float64       
 8   hightet       1554 non-null   float64       
 9   lowtet        1554 non-null   float64       
 10  closetet      1554 non-null   float64       
 11  adj_closetet  1554 non-null   float64       
 12  volumetet     1554 non-null   int64         
 13  openeth       1554 non-null   float64       
 14  higheth       1554 non-null   float64       
 15  loweth        1554 non-null   float64 

After checking, there are no null values since the non-null count for each column = the total entries i.e 1554

In [27]:
#Dropping the unnecessary columns 'closebit' and 'closeeth' since we can use 'adj_closebit' and 'adj_closeeth' which more conclusive figures.

crypto_df.drop(["closebit",  "closeeth"], axis = 1, inplace = True)
crypto_df.head(10)



datebit       openbit       highbit  ...      loweth  adj_closeeth   volumeeth
0 2017-11-21   8205.740234   8348.660156  ...  350.692993    360.401001   949912000
1 2017-11-22   8077.950195   8302.259766  ...  360.147003    380.652008   800819008
2 2017-11-23   8232.379883   8267.400391  ...  376.088013    410.165985  1845680000
3 2017-11-24   8074.020020   8374.160156  ...  402.757996    474.911011  2292829952
4 2017-11-25   8241.709961   8790.919922  ...  461.053009    466.276001  1422080000
5 2017-11-26   8789.040039   9522.929688  ...  451.605988    471.329987  1197779968
6 2017-11-27   9352.719727   9818.349609  ...  468.484985    480.355011  1396480000
7 2017-11-28   9823.429688  10125.700195  ...  466.346985    472.902008  1346499968
8 2017-11-29  10077.400391  11517.400391  ...  425.071014    427.523010  2675940096
9 2017-11-30   9906.790039  10801.000000  ...  401.243011    447.114014  1903040000

[10 rows x 17 columns]

In [28]:
#Check Consistency and remove duplicated date records
crypto_df.drop_duplicates(subset=['datebit'])

datebit       openbit  ...  adj_closeeth    volumeeth
0    2017-11-21   8205.740234  ...    360.401001    949912000
1    2017-11-22   8077.950195  ...    380.652008    800819008
2    2017-11-23   8232.379883  ...    410.165985   1845680000
3    2017-11-24   8074.020020  ...    474.911011   2292829952
4    2017-11-25   8241.709961  ...    466.276001   1422080000
...         ...           ...  ...           ...          ...
1549 2022-02-17  43937.070313  ...   2881.481934  15860206214
1550 2022-02-18  40552.132813  ...   2785.727539  15748173433
1551 2022-02-19  40026.023438  ...   2763.701172   9774183169
1552 2022-02-20  40118.101563  ...   2628.648438  11641437834
1553 2022-02-21  38339.218750  ...   2701.658936  15016932352

[1554 rows x 17 columns]

In [29]:
#Exporting our new clean and merged dataset to a single csv file.
crypto_df.to_csv("crypto_df.csv", index=False)

In [30]:
# Lets read our dataset from our CSV file and create a dataframe that we will use
crypto_df=pd.read_csv('crypto_df.csv')

In [31]:
# Let's Preview the first 10 records of our cleaned DataFrame 

crypto_df.head(10)

datebit       openbit       highbit  ...      loweth  adj_closeeth   volumeeth
0  2017-11-21   8205.740234   8348.660156  ...  350.692993    360.401001   949912000
1  2017-11-22   8077.950195   8302.259766  ...  360.147003    380.652008   800819008
2  2017-11-23   8232.379883   8267.400391  ...  376.088013    410.165985  1845680000
3  2017-11-24   8074.020020   8374.160156  ...  402.757996    474.911011  2292829952
4  2017-11-25   8241.709961   8790.919922  ...  461.053009    466.276001  1422080000
5  2017-11-26   8789.040039   9522.929688  ...  451.605988    471.329987  1197779968
6  2017-11-27   9352.719727   9818.349609  ...  468.484985    480.355011  1396480000
7  2017-11-28   9823.429688  10125.700195  ...  466.346985    472.902008  1346499968
8  2017-11-29  10077.400391  11517.400391  ...  425.071014    427.523010  2675940096
9  2017-11-30   9906.790039  10801.000000  ...  401.243011    447.114014  1903040000

[10 rows x 17 columns]

In [32]:
# Let's Preview the last 10 records of our cleaned DataFrame 

crypto_df.tail(10)

datebit       openbit  ...  adj_closeeth    volumeeth
1544  2022-02-12  42412.300781  ...   2917.362793  11254355757
1545  2022-02-13  42236.566406  ...   2883.463379   9054963563
1546  2022-02-14  42157.398438  ...   2933.479004  12164552172
1547  2022-02-15  42586.464844  ...   3179.877197  13921257873
1548  2022-02-16  44578.277344  ...   3127.830078  12352406833
1549  2022-02-17  43937.070313  ...   2881.481934  15860206214
1550  2022-02-18  40552.132813  ...   2785.727539  15748173433
1551  2022-02-19  40026.023438  ...   2763.701172   9774183169
1552  2022-02-20  40118.101563  ...   2628.648438  11641437834
1553  2022-02-21  38339.218750  ...   2701.658936  15016932352

[10 rows x 17 columns]

#3.0 DATA ANALYSIS

From this dataset, we want to able to derive meaningful Analysis that will guide the decision of a future Cryptocurrency Invest. As such, below is our Analysis.

## Annual Performance

### 1.What has been the Highest Price of the two currencies in comparison (Bitcoin and Etherium) over the 5 years?

In [33]:
# Let's find the 10 most highest prices with their corresponding dates for the Bitcoin trade over the last 5 years

bitcoin_highest_trade=crypto_df[(crypto_df['datebit'] <= '21-02-2022') & (crypto_df['datebit'] >= '09-11-2017')].nlargest(10, ['highbit'])[['datebit','highbit']]
bitcoin_highest_trade

datebit       highbit
1450  2021-11-10  68789.625000
1449  2021-11-09  68530.335938
1448  2021-11-08  67673.742188
1429  2021-10-20  66930.390625
1430  2021-10-21  66600.546875
1455  2021-11-15  66281.570313
1451  2021-11-11  65579.015625
1454  2021-11-14  65495.179688
1452  2021-11-12  65460.816406
1453  2021-11-13  64915.675781

In [34]:
# Let's find the 10 most highest prices with their corresponding dates for the Etherium trade over the last 5 years

etherium_highest_trade=crypto_df[(crypto_df['datebit'] <= '21-02-2022') & (crypto_df['datebit'] >= '09-11-2017')].nlargest(10, ['higheth'])[['datebit','higheth']]
etherium_highest_trade

datebit      higheth
1456  2021-11-16  4891.704590
1450  2021-11-10  4859.502930
1449  2021-11-09  4837.589355
1448  2021-11-08  4822.363281
1452  2021-11-12  4808.738770
1471  2021-12-01  4780.732422
1451  2021-11-11  4778.059082
1455  2021-11-15  4764.636230
1470  2021-11-30  4753.266113
1453  2021-11-13  4702.115234

In [35]:
# Let's find the 10 most highest prices with their corresponding dates for the Tether trade over the last 5 years

tether_highest_trade=crypto_df[(crypto_df['datebit'] <= '21-02-2022') & (crypto_df['datebit'] >= '09-11-2017')].nlargest(10, ['hightet'])[['datebit','hightet']]
tether_highest_trade

datebit   hightet
22   2017-12-13  1.088000
843  2020-03-13  1.084802
21   2017-12-12  1.081670
842  2020-03-12  1.078518
16   2017-12-07  1.076950
57   2018-01-17  1.069890
33   2017-12-24  1.069220
901  2020-05-10  1.060485
31   2017-12-22  1.059090
17   2017-12-08  1.057660

In [36]:
# Let's find the 10 most highest prices with their corresponding dates for the Etherium trade over the last 5 years

etherium_highest_trade=crypto_df[(crypto_df['datebit'] <= '21-02-2022') & (crypto_df['datebit'] >= '09-11-2017')].nlargest(10, ['higheth'])[['datebit','higheth']]
etherium_highest_trade

datebit      higheth
1456  2021-11-16  4891.704590
1450  2021-11-10  4859.502930
1449  2021-11-09  4837.589355
1448  2021-11-08  4822.363281
1452  2021-11-12  4808.738770
1471  2021-12-01  4780.732422
1451  2021-11-11  4778.059082
1455  2021-11-15  4764.636230
1470  2021-11-30  4753.266113
1453  2021-11-13  4702.115234

In [38]:
tether_highest_trade=crypto_df[(crypto_df['datebit'] <= '21-02-2022') & (crypto_df['datebit'] >= '09-11-2017')].nlargest(10, ['highbit'])[['datebit','hightet']]
tether_highest_trade

datebit   hightet
1450  2021-11-10  1.001987
1449  2021-11-09  1.001228
1448  2021-11-08  1.001587
1429  2021-10-20  1.000689
1430  2021-10-21  1.000436
1455  2021-11-15  1.001988
1451  2021-11-11  1.001997
1454  2021-11-14  1.001952
1452  2021-11-12  1.001994
1453  2021-11-13  1.001993

###2.What has been the Lowest Price of the two currencies in comparison (Bitcoin and Etherium)

In [39]:
# Let's find the 10 most lowest prices with their corresponding dates for the Bitcoin trade over the last 5 years

bitcoin_lowest_trade=crypto_df[(crypto_df['datebit'] <= '21-02-2022') & (crypto_df['datebit'] >= '09-11-2017')].nsmallest(10, ['lowbit'])[['datebit','lowbit']]
bitcoin_lowest_trade

datebit       lowbit
389  2018-12-15  3191.303467
388  2018-12-14  3206.542236
390  2018-12-16  3233.819824
391  2018-12-17  3253.123047
381  2018-12-07  3280.228760
387  2018-12-13  3298.132080
382  2018-12-08  3350.650635
444  2019-02-08  3391.023682
385  2018-12-11  3392.250000
443  2019-02-07  3394.218506

In [40]:
# Let's find the 10 most lowest prices with their corresponding dates for the Etherium trade over the last 5 years

etherium_lowest_trade=crypto_df[(crypto_df['datebit'] <= '21-02-2022') & (crypto_df['datebit'] >= '09-11-2017')].nsmallest(10, ['loweth'])[['datebit','loweth']]
etherium_lowest_trade

datebit     loweth
389  2018-12-15  82.829887
388  2018-12-14  83.226479
381  2018-12-07  83.469719
390  2018-12-16  84.469406
391  2018-12-17  85.382111
387  2018-12-13  85.518311
382  2018-12-08  86.825668
385  2018-12-11  87.688408
386  2018-12-12  88.263557
384  2018-12-10  90.654251

In [41]:
# Let's find the 10 most lowest prices with their corresponding dates for the Tether trade over the last 5 years

tether_lowest_trade=crypto_df[(crypto_df['datebit'] <= '21-02-2022') & (crypto_df['datebit'] >= '09-11-2017')].nsmallest(10, ['lowtet'])[['datebit','lowtet']]
tether_lowest_trade

datebit    lowtet
843  2020-03-13  0.899490
73   2018-02-02  0.925144
328  2018-10-15  0.925284
76   2018-02-05  0.944777
348  2018-11-04  0.945723
364  2018-11-20  0.950014
849  2020-03-19  0.953586
72   2018-02-01  0.954771
847  2020-03-17  0.955483
8    2017-11-29  0.955618

###3.What has been the Highest Volume traded for the two currencies in comparison (Bitcoin and Etherium) over the past 5 years

In [42]:
# Let's find the 10 most highest volumes with their corresponding dates for the Bitcoin trade over the last 5 years

bitcoin_highest_volume=crypto_df[(crypto_df['datebit'] <= '21-02-2022') & (crypto_df['datebit'] >= '09-11-2017')].nlargest(10, ['volumebit'])[['datebit','volumebit']]
bitcoin_highest_volume

datebit     volumebit
1193  2021-02-26  350967941479
1275  2021-05-19  126358098747
1147  2021-01-11  123320567399
1165  2021-01-29  117894572511
1190  2021-02-23  106102492824
1175  2021-02-08  101467222687
1244  2021-04-18   97468872758
1269  2021-05-13   96721152926
1189  2021-02-22   92052420332
1176  2021-02-09   91809846886

In [43]:
# Let's find the 10 most highest volumes with their corresponding dates for the Etherium trade over the last 5 years

etherium_highest_volume=crypto_df[(crypto_df['datebit'] <= '21-02-2022') & (crypto_df['datebit'] >= '09-11-2017')].nlargest(10, ['volumeeth'])[['datebit','volumeeth']]
etherium_highest_volume

datebit    volumeeth
1275  2021-05-19  84482912776
1269  2021-05-13  78398214539
1268  2021-05-12  69023382175
1276  2021-05-20  67610826680
1266  2021-05-10  62691789007
1260  2021-05-04  62402045158
1147  2021-01-11  60733630300
1140  2021-01-04  56945985763
1279  2021-05-23  56005721977
1249  2021-04-23  55413933925

In [44]:

tether_highest_volume=crypto_df[(crypto_df['datebit'] <= '21-02-2022') & (crypto_df['datebit'] >= '09-11-2017')].nlargest(10, ['volumetet'])[['datebit','volumetet']]
tether_highest_volume

datebit     volumetet
1275  2021-05-19  279067455600
1269  2021-05-13  238503239903
1242  2021-04-16  232378434390
1244  2021-04-18  226468468195
1262  2021-05-06  209142160582
1276  2021-05-20  203389167819
1249  2021-04-23  200928310345
1266  2021-05-10  196654261473
1190  2021-02-23  195964923867
1263  2021-05-07  195953903780

###4.What has been the Lowest Volume traded for the two currencies in comparison (Bitcoin and Etherium) over the past 5 years

In [45]:
# Let's find the 10 most lowest volumes with their corresponding dates for the Bitcoin trade over the last 5 years

bitcoin_lowest_volume=crypto_df[(crypto_df['datebit'] <= '21-02-2022') & (crypto_df['datebit'] >= '09-11-2017')].nsmallest(10, ['volumebit'])[['datebit','volumebit']]
bitcoin_lowest_volume

datebit   volumebit
235  2018-07-14  2923670016
326  2018-10-13  3064030000
327  2018-10-14  3085320000
208  2018-06-17  3104019968
207  2018-06-16  3194170112
298  2018-09-15  3216300000
338  2018-10-25  3230550000
334  2018-10-21  3253610000
319  2018-10-06  3259740000
299  2018-09-16  3273730000

In [46]:
# Let's find the 10 most lowest volumes with their corresponding dates for the Etherium trade over the last 5 years

etherium_lowest_volume=crypto_df[(crypto_df['datebit'] <= '21-02-2022') & (crypto_df['datebit'] >= '09-11-2017')].nsmallest(10, ['volumeeth'])[['datebit','volumeeth']]
etherium_lowest_volume

datebit   volumeeth
1    2017-11-22   800819008
11   2017-12-02   943649984
138  2018-04-08   948488000
0    2017-11-21   949912000
137  2018-04-07   951475008
136  2018-04-06   967105984
12   2017-12-03   990556992
13   2017-12-04  1005550016
340  2018-10-27  1084810000
337  2018-10-24  1102220000

In [47]:
tether_lowest_volume=crypto_df[(crypto_df['datebit'] <= '21-02-2022') & (crypto_df['datebit'] >= '09-11-2017')].nsmallest(10, ['volumetet'])[['datebit','volumetet']]
tether_lowest_volume

datebit  volumetet
1   2017-11-22  499812000
0   2017-11-21  622649024
5   2017-11-26  636451968
4   2017-11-25  651987968
13  2017-12-04  668510016
11  2017-12-02  679950016
2   2017-11-23  735883008
10  2017-12-01  749395968
6   2017-11-27  784436992
7   2017-11-28  794764032

###5.What has been the percentage increase in price for the two currencies in comparison (Bitcoin and Etherium) over the past 5 years

In [62]:
# Let's start by previewing the start and end highbit value of bitcoin in every year for the past 5 years.

range1=crypto_df[(crypto_df['datebit'] >= '2017-11-09') & (crypto_df['datebit'] <= '2018-11-09')].iloc[[0,-1]][['datebit','highbit']]
print(range1)
range2=crypto_df[(crypto_df['datebit'] > '2018-11-09') & (crypto_df['datebit'] <= '2019-11-09')].iloc[[0,-1]][['datebit','highbit']]
print(range2)
range3=crypto_df[(crypto_df['datebit'] > '2019-11-09') & (crypto_df['datebit'] <= '2020-11-09')].iloc[[0,-1]][['datebit','highbit']]
print(range3)
range4=crypto_df[(crypto_df['datebit'] > '2020-11-09') & (crypto_df['datebit'] <= '2021-11-09')].iloc[[0,-1]][['datebit','highbit']]
print(range4)
range5=crypto_df[(crypto_df['datebit'] > '2021-11-09') & (crypto_df['datebit'] <= '2022-11-21')].iloc[[0,-1]][['datebit','highbit']]
print(range5)


        datebit      highbit
0    2017-11-21  8348.660156
353  2018-11-09  6456.459961
        datebit      highbit
354  2018-11-10  6437.279785
718  2019-11-09  8891.818359
         datebit       highbit
719   2019-11-10   9103.826172
1084  2020-11-09  15785.136719
         datebit       highbit
1085  2020-11-10  15450.329102
1449  2021-11-09  68530.335938
         datebit       highbit
1450  2021-11-10  68789.625000
1553  2022-02-21  39379.757813


In [56]:
%pip install tabulate

In [57]:
from tabulate import tabulate
range1['Bitcoin Price Difference'] = range1['highbit'].diff(periods=1)

range2['Bitcoin Price Difference'] = range2['highbit'].diff(periods=1)

range3['Bitcoin Price Difference'] = range3['highbit'].diff(periods=1)

range4['Bitcoin Price Difference'] = range4['highbit'].diff(periods=1)

range5['Bitcoin Price Difference'] = range5['highbit'].diff(periods=1)

range1['Bitcoin Price % Profit/Loss'] = ((range1['Bitcoin Price Difference']/7446.83)*100)

range2['Bitcoin Price % Profit/Loss'] = ((range2['Bitcoin Price Difference']/6437.28)*100)

range3['Bitcoin Price % Profit/Loss'] = ((range3['Bitcoin Price Difference']/9103.83)*100)

range4['Bitcoin Price % Profit/Loss'] = ((range4['Bitcoin Price Difference']/15450.3)*100)

range5['Bitcoin Price % Profit/Loss'] = ((range5['Bitcoin Price Difference']/68789.63)*100)

print(tabulate(range1, headers='keys', tablefmt='psql'))
print(tabulate(range2, headers='keys', tablefmt='psql'))
print(tabulate(range3, headers='keys', tablefmt='psql'))
print(tabulate(range4, headers='keys', tablefmt='psql'))
print(tabulate(range5, headers='keys', tablefmt='psql'))

+-----+------------+-----------+----------------------------+-------------------------------+
|     | datebit    |   highbit |   Bitcoin Price Difference |   Bitcoin Price % Profit/Loss |
|-----+------------+-----------+----------------------------+-------------------------------|
|   0 | 2017-11-21 |   8348.66 |                      nan   |                      nan      |
| 353 | 2018-11-09 |   6456.46 |                    -1892.2 |                      -25.4095 |
+-----+------------+-----------+----------------------------+-------------------------------+
+-----+------------+-----------+----------------------------+-------------------------------+
|     | datebit    |   highbit |   Bitcoin Price Difference |   Bitcoin Price % Profit/Loss |
|-----+------------+-----------+----------------------------+-------------------------------|
| 354 | 2018-11-10 |   6437.28 |                     nan    |                      nan      |
| 718 | 2019-11-09 |   8891.82 |                    2454.54 

In [75]:
# previewing the start and end hightet value of Tether in every year for the past 5 years.

range_tet1=crypto_df[(crypto_df['datebit'] >= '2017-11-09') & (crypto_df['datebit'] <= '2018-11-09')].iloc[[0,-1]][['datebit','hightet']]
print(range_tet1)
range_tet2=crypto_df[(crypto_df['datebit'] > '2018-11-09') & (crypto_df['datebit'] <= '2019-11-09')].iloc[[0,-1]][['datebit','hightet']]
print(range_tet2)
range_tet3=crypto_df[(crypto_df['datebit'] > '2019-11-09') & (crypto_df['datebit'] <= '2020-11-09')].iloc[[0,-1]][['datebit','hightet']]
print(range_tet3)
range_tet4=crypto_df[(crypto_df['datebit'] > '2020-11-09') & (crypto_df['datebit'] <= '2021-11-09')].iloc[[0,-1]][['datebit','hightet']]
print(range_tet4)
range_tet5=crypto_df[(crypto_df['datebit'] > '2021-11-09') & (crypto_df['datebit'] <= '2022-11-21')].iloc[[0,-1]][['datebit','hightet']]
print(range_tet5)

        datebit   hightet
0    2017-11-21  1.011200
353  2018-11-09  0.999944
        datebit   hightet
354  2018-11-10  0.998994
718  2019-11-09  1.007227
         datebit   hightet
719   2019-11-10  1.008127
1084  2020-11-09  1.001397
         datebit   hightet
1085  2020-11-10  1.001292
1449  2021-11-09  1.001228
         datebit   hightet
1450  2021-11-10  1.001987
1553  2022-02-21  1.000874


As per the data above, all the negative values indicate a price drop, hence a loss. The positive values, on the other hand, indicate a price increase, hence a profit. This analysis is done for the opening value and closing value of Bitcoin for every year for the period of five years.

From the Analysis above, Bitcoin experience the highest profit of 344% between 10-11-2020 and 09-11-21. It can also been concluded that Bitcoin experince the greatest loss between 10-11-2021 and 21-02-2022, which translated to 43% loss.

In [58]:
# Let's start by previewing the start and end highbit value of etherium every year for the past 5 years.

range_eth1=crypto_df[(crypto_df['datebit'] <= '2018-11-09') & (crypto_df['datebit'] >= '2017-11-09')].iloc[[0,-1]][['datebit','higheth']]
print(range_eth1)
range_eth2=crypto_df[(crypto_df['datebit'] <= '2019-11-09') & (crypto_df['datebit'] > '2018-11-09')].iloc[[0,-1]][['datebit','higheth']]
print(range_eth2)
range_eth3=crypto_df[(crypto_df['datebit'] <= '2020-11-09') & (crypto_df['datebit'] > '2019-11-09')].iloc[[0,-1]][['datebit','higheth']]
print(range_eth3)
range_eth4=crypto_df[(crypto_df['datebit'] <= '2021-11-09') & (crypto_df['datebit'] > '2020-11-09')].iloc[[0,-1]][['datebit','higheth']]
print(range_eth4)
range_eth5=crypto_df[(crypto_df['datebit'] <= '2022-11-21') & (crypto_df['datebit'] > '2021-11-09')].iloc[[0,-1]][['datebit','higheth']]
print(range_eth5)

        datebit     higheth
0    2017-11-21  372.470001
353  2018-11-09  213.315994
        datebit     higheth
354  2018-11-10  213.856995
718  2019-11-09  185.838531
         datebit     higheth
719   2019-11-10  190.920609
1084  2020-11-09  457.349609
         datebit      higheth
1085  2020-11-10   453.758362
1449  2021-11-09  4837.589355
         datebit      higheth
1450  2021-11-10  4859.502930
1553  2022-02-21  2750.815918


In [77]:
from tabulate import tabulate
range_tet1['Tether Price Difference'] = range_tet1['hightet'].diff(periods=1)

range_tet2['Tether Price Difference'] = range_tet2['hightet'].diff(periods=1)

range_tet3['Tether Price Difference'] = range_tet3['hightet'].diff(periods=1)

range_tet4['Tether Price Difference'] = range_tet4['hightet'].diff(periods=1)

range_tet5['Tether Price Difference'] = range_tet5['hightet'].diff(periods=1)

range_tet1['Tether Price % Profit/Loss'] = ((range_tet1['Tether Price Difference']/329.45)*100)

range_tet2['Tether Price % Profit/Loss'] = ((range_tet2['Tether Price Difference']/213.86)*100)

range_tet3['Tether Price % Profit/Loss'] = ((range_tet3['Tether Price Difference']/190.92)*100)

range_tet4['Tether Price % Profit/Loss'] = ((range_tet4['Tether Price Difference']/453.76)*100)

range_tet5['Tether Price % Profit/Loss'] = ((range_tet5['Tether Price Difference']/4859.50)*100)


print(tabulate(range_tet1, headers='keys', tablefmt='psql'))
print(tabulate(range_tet2, headers='keys', tablefmt='psql'))
print(tabulate(range_tet3, headers='keys', tablefmt='psql'))
print(tabulate(range_tet4, headers='keys', tablefmt='psql'))
print(tabulate(range_tet5, headers='keys', tablefmt='psql'))

+-----+------------+-----------+---------------------------+------------------------------+
|     | datebit    |   hightet |   Tether Price Difference |   Tether Price % Profit/Loss |
|-----+------------+-----------+---------------------------+------------------------------|
|   0 | 2017-11-21 |  1.0112   |                nan        |                  nan         |
| 353 | 2018-11-09 |  0.999944 |                 -0.011256 |                   -0.0034166 |
+-----+------------+-----------+---------------------------+------------------------------+
+-----+------------+-----------+---------------------------+------------------------------+
|     | datebit    |   hightet |   Tether Price Difference |   Tether Price % Profit/Loss |
|-----+------------+-----------+---------------------------+------------------------------|
| 354 | 2018-11-10 |  0.998994 |              nan          |                 nan          |
| 718 | 2019-11-09 |  1.00723  |                0.00823295 |                   0

As per the data above, all the negative values indicate a price drop, hence a loss. The positive values, on the other hand, indicate a price increase, hence a profit. This analysis is done for the opening value and closing value of Etherium for every year for the period of five years.

From the Analysis above, Etherium experience the highest profit of 966% between 10-11-2020 and 09-11-21. It can also been concluded that Etherium experince the greatest loss between 10-11-2021 and 21-02-2022, which translated to 43% loss.

In [60]:
import plotly.express as px
import plotly.graph_objects as go

fig = go.Figure()
fig.add_trace(go.Scatter(x=crypto_df.datebit, y=crypto_df.adj_closebit, name='Closing Price', line=dict(color='orange', width=2)))
fig.update_layout(title=('Bitcoin Price'),
                    xaxis_title='Year',
                   yaxis_title='Closing Price')
fig.show()

In [61]:
fig1 = go.Figure()
fig1.add_trace(go.Scatter(x=crypto_df.datebit, y=crypto_df.adj_closeeth, name='Closing Price', line=dict(color='orange', width=2)))
fig1.update_layout(title=('Etherium Price'),
                    xaxis_title='Year',
                   yaxis_title='Closing Price')
fig1.show()

In [78]:
import plotly.express as px
import plotly.graph_objects as go

fig = go.Figure()
fig.add_trace(go.Scatter(x=crypto_df.datebit, y=crypto_df.adj_closetet, name='Closing Price', line=dict(color='orange', width=2)))
fig.update_layout(title=('Tether Price'),
                    xaxis_title='Year',
                   yaxis_title='Closing Price')
fig.show()

## 2018 Crash

In [79]:
import plotly.express as px
import plotly.graph_objects as go

In [81]:

def PlottingFunctionThree(x,y1,y2,y3,name1,name2,name3, color1, color2, color3, title,xtitle,ytitle):
    fig = go.Figure()
    fig.add_trace(go.Scatter(x= x, y= y1, name= name1, line=dict(color= color1, width=2)))
    fig.add_trace(go.Scatter(x= x, y= y2, name= name2, line=dict(color= color2, width=2)))
    fig.add_trace(go.Scatter(x= x, y= y3, name= name3, line=dict(color= color3, width=2)))
    fig.update_layout(title = title,
            xaxis_title=xtitle,
            yaxis_title= ytitle)
    fig.show()

In [88]:
crashperiod = crypto_df[(crypto_df['datebit'] > '2017-09-01') & (crypto_df['datebit'] < '2018-12-31')]
PlottingFunctionThree(crashperiod.datebit, crashperiod.adj_closebit, crashperiod.adj_closeeth, crashperiod.adj_closetet, 'Bitcoin', 'Ethereum', 'Tether', 'orange', 'purple','green',
 '2018 Crash Performance', 'Date', 'Closing Price')

## Pre-Covid

In [90]:
precovidperiod = crypto_df[(crypto_df['datebit'] > '2017-09-01') & (crypto_df['datebit'] < '2018-12-31')]
PlottingFunctionThree(crashperiod.datebit, precovidperiod.adj_closebit, precovidperiod.adj_closeeth, precovidperiod.adj_closetet, 'Bitcoin', 'Ethereum', 'Tether', 'orange', 'purple', 'green',
 'Pre-Covid Performance', 'Date', 'Closing Price')

## Covid Period

In [92]:
covidperiod = crypto_df[(crypto_df['datebit'] > '2020-01-30') & (crypto_df['datebit'] < '2021-01-01')]
PlottingFunctionThree(covidperiod.datebit, covidperiod.adj_closebit, covidperiod.adj_closeeth, covidperiod.adj_closetet, 'Bitcoin', 'Ethereum', 'Tether', 'orange', 'purple','green',
 'Pre-Covid Performance', 'Date', 'Closing Price')

## Post Covid

In [95]:
postcovidperiod = crypto_df[crypto_df['datebit'] > '2021-01-01']
PlottingFunctionThree(postcovidperiod.datebit, postcovidperiod.adj_closebit, postcovidperiod.adj_closeeth, postcovidperiod.adj_closetet, 'Bitcoin', 'Ethereum', 'Tether', 'orange', 'purple', 'green',
 'Pre-Covid Performance', 'Date', 'Closing Price')